tests/test_fips.py

```diff
@@ -152,3 +186,57 @@ def fips_mode_setup_check(container_per_test: ContainerData) -> None:
 )
 def test_openssl_fips_hashes(container_per_test: ContainerData):
     openssl_fips_hashes_test_fnct(container_per_test)
+
+
+@pytest.mark.parametrize(
+    "container_per_test", FIPS_GNUTLS_TESTER_IMAGES, indirect=True
+)
+def test_gnutls_binary(container_per_test: ContainerData) -> None:
+    """Check that a binary linked against Gnutls obeys the host's FIPS mode
+    setting:
+
+    - build a container image using :py:const:`DOCKERFILE_GNUTLS`
+    - run the bundled binary compiled from :file:`tests/files/fips-gnutls-test.c` with
+      all FIPS digests and assert that it successfully calculates the message
+      digest
+    - rerun the same binary with non-FIPS digests and assert that this fails
+      with the expected error message.
+
+    """
+
+    rr = container_per_test.connection.check_output(
+        
```

Comment:
```suggestion
```

Reply:
code is outdated(this section was added for debugging)